%YAML:1.0

#common parameters
savePath: "/home/ron/SensorFusion/src/fusion/config/log/"
bag_path: "/home/ron/SensorFusion/src/fusion/config/camd435i.bag"

scan_topic_name: "/scan"
img_topic_name: "/camera/color/image_raw"
depth_topic_name: "/camera/aligned_depth_to_color/image_raw"

# tag info
tag_type: 1 # 1: kalibr_tag, 2: apriltag
tag_size: 0.088 # tag size, unit meter
tag_spacing: 0.3 # tag spacing, only used in kalibr_tag. For details, please see kalibr tag description.
black_border: 2 # if you use kalibr_tag black_boarder = 2; if you use apriltag black_boarder = 1
 
#camera calibration 
model_type: PINHOLE
camera_name: camera
image_width: 1280
image_height: 720

distortion_parameters:
   k1: 0.14408587
   k2: -0.09078564
   p1: 0.00510793
   p2: 0.01576679
projection_parameters:
   fx: 946.87343919
   fy: 945.06707689
   cx: 677.78046819
   cy: 367.33856022




# Camera calibration and distortion parameters (OpenCV) 
Camera.fx: 611.6619978415015
Camera.fy: 612.3504421750998
Camera.cx: 321.2198443262521
Camera.cy: 231.62284705305117

Camera.k1: 0.16375563343621427
Camera.k2: -0.3102768678407425
Camera.p1: -0.003912176364130468
Camera.p2: 0.00035347945978329463

Camera.width: 1280
Camera.height: 720

# Camera frames per second 
Camera.fps: 30.0

# IR projector baseline times fx (aprox.)
Camera.bf: 40.0

# Color order of the images (0: BGR, 1: RGB. It is ignored if images are grayscale)
Camera.RGB: 1

# Close/Far threshold. Baseline times.
ThDepth: 40.0

# Deptmap values factor
DepthMapFactor: 500.0

#--------------------------------------------------------------------------------------------
# ORB Parameters
#--------------------------------------------------------------------------------------------

# ORB Extractor: Number of features per image
ORBextractor.nFeatures: 500

# ORB Extractor: Scale factor between levels in the scale pyramid 	
ORBextractor.scaleFactor: 1.2

# ORB Extractor: Number of levels in the scale pyramid	
ORBextractor.nLevels: 8

# ORB Extractor: Fast threshold
# Image is divided in a grid. At each cell FAST are extracted imposing a minimum response.
# Firstly we impose iniThFAST. If no corners are detected we impose a lower value minThFAST
# You can lower these values if your images have low contrast			
ORBextractor.iniThFAST: 20
ORBextractor.minThFAST: 7

#--------------------------------------------------------------------------------------------
# Viewer Parameters
#--------------------------------------------------------------------------------------------
Viewer.KeyFrameSize: 0.05
Viewer.KeyFrameLineWidth: 1
Viewer.GraphLineWidth: 0.9
Viewer.PointSize:2
Viewer.CameraSize: 0.08
Viewer.CameraLineWidth: 3
Viewer.ViewpointX: 0
Viewer.ViewpointY: -0.7
Viewer.ViewpointZ: -1.8
Viewer.ViewpointF: 500

